plasmapy/simulation/particle_tracker.py

```diff
@@ -169,6 +170,89 @@ def total(self) -> float:
         return float(self.tracker.nparticles)
 
 
+class ParticlesPushedThroughGridsTerminationCondition(AbstractTerminationCondition):
+    """Termination condition corresponding to stopping the simulation when a provided
+    proportion of particles have entered and exited the grids.
+
+    Parameters
+    ----------
+    fraction_entered_threshold: float, optional
+        The proportion of particles that must enter the grids to terminate the simulation.
+
+    fraction_exited_threshold: float, optional
+        The proportion of particles that must leave the grids to terminate the simulation.
+        This does not include particles that have never entered the grids.
+    """
+
+    def __init__(
+        self,
+        fraction_entered_threshold: float | None,
+        fraction_exited_threshold: float | None,
+    ):
+        super().__init__()
+
+        self.fraction_entered_threshold = fraction_entered_threshold
+        self.fraction_exited_threshold = fraction_exited_threshold
+
+    @property
+    def require_synchronized_dt(self) -> bool:
+        """The termination condition does not require a synchronized time step."""
+        return False
+
+    @property
+    def progress_description(self) -> str:
+        """The termination condition tracks the number of particles remaining in the grids."""
+        return "Fraction of particles remaining"
+
+    @property
+    def units_string(self) -> str:
+        """The termination condition tracks particles."""
+
+        return "Particles"
+
+    @property
+    def is_finished(self) -> bool:
+        r"""
+        Check to see if the proportion of particles that have entered and exited the grid meet thresholds.
+        """
+
+        # Of the particles that have entered the grid, how many are currently
+        # on the grid?
+        # if/else avoids dividing by zero
+        if self._particle_tracker.num_entered > 0:
+            # Normalize to the number that have entered a grid
+            still_on = (
+                np.sum(self._particle_tracker.on_any_grid)
+                / self._particle_tracker.num_entered
+            )
+        else:
+            still_on = 0.0
+
+        if (
```

Comment:
See above, also `fraction_exited_threshold` either needs a default or to be required - otherwise this terminates immediately?